skaffold/src/test/kotlin/com/google/container/tools/skaffold/run/ui/SkaffoldProfilesComboBoxTest.kt

```diff
@@ -0,0 +1,220 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.container.tools.skaffold.run.ui
+
+import com.google.common.truth.Truth.assertThat
+import com.google.container.tools.test.ContainerToolsRule
+import com.google.container.tools.test.UiTest
+import com.intellij.mock.MockVirtualFile
+import org.junit.Rule
+import org.junit.Test
+
+/** Unit tests for [SkaffoldProfilesComboBox] */
+class SkaffoldProfilesComboBoxTest {
+    @get:Rule
+    val containerToolsRule = ContainerToolsRule(this)
+
+    @Test
+    @UiTest
+    fun `no skaffold yaml file results in empty disabled profiles combobox`() {
+        val profilesComboBox = SkaffoldProfilesComboBox()
+
+        assertThat(profilesComboBox.model.size).isEqualTo(0)
+        assertThat(profilesComboBox.isEnabled).isFalse()
+    }
+
+    @Test
+    @UiTest
+    fun `no skaffold yaml file results in null profile selection`() {
+        val profilesComboBox = SkaffoldProfilesComboBox()
+
+        assertThat(profilesComboBox.getSelectedProfile()).isNull()
+    }
+
+    @Test
+    @UiTest
+    fun `skaffold yaml without profiles results in disabled combobox with default profile`() {
+        val skaffoldYamlFile = MockVirtualFile.file("skaffold.yaml")
+        skaffoldYamlFile.setText(
+            """
+            apiVersion: skaffold/v1beta1
+            kind: Config
+            build:
+                artifact:
+                - image: docker.io/local-image
+
+        """
+        )
+        val profilesComboBox = SkaffoldProfilesComboBox()
+
+        profilesComboBox.skaffoldFileUpdated(skaffoldYamlFile)
+
+        assertThat(profilesComboBox.model.size).isEqualTo(1)
+        assertThat(profilesComboBox.model.getElementAt(0)).isEqualTo("default")
+        assertThat(profilesComboBox.isEnabled).isFalse()
+    }
+
+    @Test
+    @UiTest
+    fun `skaffold yaml without profiles results in null profile selection`() {
+        val skaffoldYamlFile = MockVirtualFile.file("skaffold.yaml")
+        skaffoldYamlFile.setText(
+            """
+            apiVersion: skaffold/v1beta1
+            kind: Config
+            build:
+                artifact:
+                - image: docker.io/local-image
+
+        """
+        )
+        val profilesComboBox = SkaffoldProfilesComboBox()
+
+        profilesComboBox.skaffoldFileUpdated(skaffoldYamlFile)
+
+        assertThat(profilesComboBox.getSelectedProfile()).isNull()
+    }
+
+    @Test
+    @UiTest
+    fun `skaffold yaml with profiles results in combobox with valid profile selection`() {
```

Comment:
I guess implicit in here is that when there are multiple profiles, then "default" will be selected by default?

Reply:
Hmm.. not really a purpose of this test case to test what is the resulting selection, but rather a list of profiles to choose from is correct. How about
`skaffold yaml with profiles produces valid list of profiles to select from`

Reply:
wording sgtm. Is there another test case to test which selection is defaulted? (sorry if you have one already and I missed it).

Reply:
There is one for no profiles yaml file. probably makes sense to add one for one with profiles too to ensure proper default selection. Adding now.

Reply:
done.